frontend/lib/create_listing/view/create_listing.view.dart

```diff
@@ -0,0 +1,414 @@
+            if (isPrice) {
+              final price = double.tryParse(value);
+              if (price == null) {
+                return 'Enter a price to continue.';
+              }
+              if (price <= 0) {
+                return 'Price must be greater than 0';
+              }
+              if (price > CreateListingModel.maxPrice) {
+                return 'Price must be less than \$${CreateListingModel.maxPrice}';
+              }
+            }
+            return null;
+          },
+          decoration: InputDecoration(
+            errorStyle: const TextStyle(
+              color: Color(0xFFFF5252),
+              fontSize: 12,
+              height: 0.8,
+            ),
+            hintText: label,
+            hintStyle: TextStyle(
+              color: Colors.grey[600],
+            ),
+            filled: true,
+            fillColor: Colors.grey[200],
+            border: OutlineInputBorder(
+              borderRadius: BorderRadius.circular(12),
+              borderSide: BorderSide.none,
+            ),
+            errorBorder: OutlineInputBorder(
+              borderRadius: BorderRadius.circular(12),
+              borderSide: const BorderSide(color: Color(0xFFFF5252), width: 1),
+            ),
+            focusedErrorBorder: OutlineInputBorder(
+              borderRadius: BorderRadius.circular(12),
+              borderSide: const BorderSide(color: Color(0xFFFF5252), width: 1),
+            ),
+            contentPadding: const EdgeInsets.symmetric(
+              horizontal: 16,
+              vertical: 14,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildConditionSelector() {
+    return Consumer<CreateListingViewModel>(
+      builder: (_, model, __) {
+        final bool showError = _showValidationErrors && model.condition.isEmpty;
+        
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            const Text(
+              'Condition',
+              style: TextStyle(
+                fontSize: 16,
+                fontWeight: FontWeight.w500,
+              ),
+            ),
+            const SizedBox(height: 8),
+            Row(
+              children: [
+                _buildConditionChip('New', model),
+                _buildConditionChip('Almost New', model),
+                _buildConditionChip('Used', model),
+                _buildConditionChip('Fair', model),
+              ],
+            ),
+            if (showError)
+              const Padding(
+                padding: EdgeInsets.only(top: 4),
+                child: Text(
+                  'Select a condition to continue.',
+                  style: TextStyle(
+                    color: Color(0xFFFF5252),
+                    fontSize: 12,
+                  ),
+                ),
+              ),
+          ],
+        );
+      },
+    );
+  }
+
+  Widget _buildConditionChip(String condition, CreateListingViewModel model) {
```

Comment:
This can move to the definition of the Condition selector or as a separate component, you can decide what's best for this depending on the code you see after refactoring.